packages/utilities/src/bn.ts

```diff
@@ -0,0 +1,301 @@
+    )!
+  }
+
+  static normalizeDecimals(...args: BN[]): BN[] {
+    const numWithBiggestDecimals = BN.getBiggestDecimal(...args)
+
+    return args.map(el =>
+      BN.fromFraction(
+        el!.bn.multipliedBy(
+          BN.#instance(10).pow(
+            numWithBiggestDecimals!.cfg.decimals - el.cfg.decimals,
+          ),
+        ),
+        numWithBiggestDecimals!.cfg.decimals,
+      ),
+    )
+  }
+
+  static min(...args: BN[]): BN {
+    const numWithBiggestDecimals = BN.getBiggestDecimal(...args)
+
+    return new BN(
+      BigNumber.minimum(...BN.normalizeDecimals(...args).map(el => el.#bn)),
+      {
+        decimals: numWithBiggestDecimals.cfg.decimals,
+      },
+    )
+  }
+
+  static max(...args: BN[]): BN {
+    const numWithBiggestDecimals = BN.getBiggestDecimal(...args)
+
+    return new BN(
+      BigNumber.maximum(...BN.normalizeDecimals(...args).map(el => el.#bn)),
+      {
+        decimals: numWithBiggestDecimals.cfg.decimals,
+      },
+    )
+  }
+
+  mul(other: BN): BN {
+    const numWithBiggestDecimals = BN.getBiggestDecimal(this, other)
+    const [numA, numB] = BN.normalizeDecimals(this, other)
+
+    return new BN(
+      numA.bn
+        .multipliedBy(numB.bn)
+        .dividedBy(BN.#instance(10).pow(numWithBiggestDecimals.cfg.decimals)),
+      {
+        ...this.#cfg,
+        decimals: numWithBiggestDecimals.cfg.decimals,
+      },
+    )
+  }
+
+  div(other: BN): BN {
+    if (other.bn.isZero()) throw new Error('Division by zero')
+
+    const numWithBiggestDecimals = BN.getBiggestDecimal(this, other)
+    const [numA, numB] = BN.normalizeDecimals(this, other)
+
+    return new BN(numA.bn.dividedBy(numB.bn), {
+      ...this.#cfg,
+      decimals: numWithBiggestDecimals.cfg.decimals,
+    })
+  }
+
+  add(other: BN): BN {
+    const numWithBiggestDecimals = BN.getBiggestDecimal(this, other)
+    const [numA, numB] = BN.normalizeDecimals(this, other)
+
+    return new BN(numA.bn.plus(numB.bn), {
+      ...this.#cfg,
+      decimals: numWithBiggestDecimals.cfg.decimals,
+    })
+  }
+
+  sub(other: BN): BN {
+    const numWithBiggestDecimals = BN.getBiggestDecimal(this, other)
+    const [numA, numB] = BN.normalizeDecimals(this, other)
+
+    return new BN(numA.bn.minus(numB.bn), {
+      ...this.#cfg,
+      decimals: numWithBiggestDecimals.cfg.decimals,
+    })
+  }
+
+  pow(other: number): BN {
+    return new BN(
+      this.#bn
+        .pow(BN.#instance(other))
+        .dividedBy(
+          BN.#instance(10).pow(
+            BN.#instance(this.#cfg.decimals).multipliedBy(
+              BN.#instance(other - 1),
+            ),
+          ),
+        ),
+      this.#cfg,
+    )
+  }
+
+  /**
+   * this > other => 1;
+   * this < other => -1;
+   * this === other => 0;
+   *
+   * @param {BnLike} other
+   * @returns {number}
+   */
+  compare(other: BN): number {
+    const [numA, numB] = BN.normalizeDecimals(this, other)
+
+    return numA.bn.comparedTo(numB.bn)
+  }
```

Comment:
let's do this method private and expose public methods such as `greaterThan`, `isSame`, `lessThan`